test/test_connection_pool.rb

```diff
@@ -109,38 +109,39 @@ def test_with
     assert Thread.new { pool.checkout }.join
   end
 
-  def test_with_with_dangerous_timeouts
-    case RUBY_ENGINE.to_sym
-    when :jruby
-      skip('JRuby GC dislikes this test')
-    when :ruby
-      if RUBY_VERSION == '2.0.0' && RUBY_PATCHLEVEL == 598
-        skip("#{RUBY_VERSION}p#{RUBY_PATCHLEVEL} GC dislikes this test")
+  def test_with_timeout
+    pool = ConnectionPool.new(:timeout => 0, :size => 1) { Object.new }
+
+    assert_raises Timeout::Error do
+      Timeout.timeout(0.01) do
+        pool.with do |obj|
+          assert_equal 0, pool.instance_variable_get(:@available).instance_variable_get(:@que).size
+          sleep 0.011
+        end
       end
     end
+    assert_equal 1, pool.instance_variable_get(:@available).instance_variable_get(:@que).size
+  end
 
-    marker_class = Class.new
-    pool = ConnectionPool.new(:timeout => 0, :size => 1) { marker_class.new }
-
-    # no "connections" allocated yet
-    assert_equal [], ObjectSpace.each_object(marker_class).to_a
-
-    checkin_time = 0.05
+  def test_checkout_ignores_timeout
+    pool = ConnectionPool.new(:timeout => 0, :size => 1) { Object.new }
+    def pool.checkout(options)
+      obj = super
+      sleep 0.015
+      obj
+    end
 
+    did_something = false
     assert_raises Timeout::Error do
-      Timeout.timeout(checkin_time) do
-        pool.with do
-          # a "connection" has been allocated
-          refute_equal [], ObjectSpace.each_object(marker_class).to_a
-          sleep 2 * checkin_time
+      Timeout.timeout(0.01) do
+        pool.with do |obj|
+          did_something = true
+          assert_equal 0, pool.instance_variable_get(:@available).instance_variable_get(:@que).size
         end
       end
     end
-
-    GC.start
-
-    # no dangling references to this "connection" remain
-    assert_equal [], ObjectSpace.each_object(marker_class).to_a
+    assert did_something
+    assert_equal 1, pool.instance_variable_get(:@available).instance_variable_get(:@que).size
```

Comment:
this checks that the connection got checked back in, right? isn't the previous test asserting that?